CodePush.m

```diff
@@ -468,6 +529,55 @@ - (void)savePendingUpdate:(NSString *)packageHash
 }
 
 /*
+ * This method is checks if a new status update exists (new version was installed, 
+ * or an update failed) and return its details (version label, status).
+ */
+RCT_EXPORT_METHOD(getNewStatusReport:(RCTPromiseResolveBlock)resolve
+                            rejecter:(RCTPromiseRejectBlock)reject)
+{
+    // Check if the current appVersion has been reported.
+    NSString *appVersion = [[CodePushConfig current] appVersion];
+    if ([self isDeploymentStatusNotYetReported:appVersion]) {
+        [self recordDeploymentStatusReported:appVersion
+                                      status:DeploymentSucceeded];
+        resolve(@{ @"appVersion": appVersion });
+        return;
+    }
+    
+    // Check if there was a rollback that was not yet reported
+    NSUserDefaults *preferences = [NSUserDefaults standardUserDefaults];
+    NSMutableArray *failedUpdates = [preferences objectForKey:FailedUpdatesKey];
+    if (failedUpdates) {
+        NSDictionary* lastFailedPackage = [failedUpdates lastObject];
+        if (lastFailedPackage) {
+            NSString* lastFailedPackageIdentifier = [self getPackageStatusReportIdentifier:lastFailedPackage];
+            if (lastFailedPackageIdentifier && [self isDeploymentStatusNotYetReported:lastFailedPackageIdentifier]) {
+                [self recordDeploymentStatusReported:lastFailedPackageIdentifier
+                                              status:DeploymentFailed];
+                resolve(@{ @"package": lastFailedPackage, @"status": DeploymentFailed });
+                return;
+            }
+        }
+    }
+    
+    // Check if the current CodePush package has been reported
+    NSError *error;
+    NSDictionary* currentPackage = [CodePushPackage getCurrentPackage:&error];
+    if (currentPackage) {
```

Comment:
Could notifyApplicationReady be called multiple times? Suppose the developer puts notifyApplicationReady in componentDidMount() to one of the components in the app, perhaps the main page. If the user then switches to a different page/component and comes back later, isFirstRun would still evaluate to true.